src/arch/linux/mapping/mapfile.c

```diff
@@ -124,8 +125,10 @@ static int open_mapping_f(int cap)
     /* /dev/shm may be mounted noexec, and then mounting PROT_EXEC fails.
        However mprotect may work around this (maybe not in future kernels)
     */
-    mpool = mmap(0, mapsize, PROT_READ|PROT_WRITE,
-    		MAP_SHARED, tmpfile_fd, 0);
+    mpool = mmap_mapping_huge_page_aligned(cap, mapsize, PROT_NONE);
```

Comment:
I wonder why its a good idea.
Putting shm in aligned block doesn't
mean every other alloc will be similar
aligned.
As such, I am not sure this patch is
needed as a whole.

Reply:
True, I just did that to waste a bit less space. This way the lowmem_base alloc can go right at the beginning, then a little hole, then the vgaemu map, and the rest (EMS/XMS) is allocated on demand between the first two or after vgaemu. But it may not matter all that much in the end, as EMS/XMS can also be allocated before lowmem, if it's not aligned here.